UnityProject/Assets/Scripts/Managers/NetworkManagement/SpawnHandler.cs

```diff
@@ -12,60 +12,69 @@ public static void SpawnViewer(NetworkConnection conn, short playerControllerId,
-		PlayerList.Instance.UpdatePlayer(conn, dummyPlayer);
-		NetworkServer.AddPlayerForConnection(conn, dummyPlayer, 0);
-		if (connectedPlayer.Script.PlayerSync != null) {
-			connectedPlayer.Script.PlayerSync.NotifyPlayers(true);
-		}
+		TransferPlayer(null, 0, dummyPlayer, null, EVENT.PlayerSpawned);
 	}
 
-	public static void RespawnPlayer(NetworkConnection conn, short playerControllerId, JobType jobType)
+	public static void RespawnPlayer(NetworkConnection conn, short playerControllerId, JobType jobType, GameObject oldBody)
 	{
 		GameObject player = CreatePlayer(jobType);
-		TransferPlayer(conn, playerControllerId, player);
+		TransferPlayer(conn, playerControllerId, player, oldBody, EVENT.PlayerSpawned);
+	}
+
+	public static void SpawnPlayerGhost(NetworkConnection conn, short playerControllerId, GameObject oldBody)
+	{
+		GameObject ghost = CreateGhost(oldBody);
+		TransferPlayer(conn, playerControllerId, ghost, oldBody, EVENT.GhostSpawned);
 	}
 
 	/// <summary>
-	/// Spawn the ghost as a new object and set that as the focus / what the user controls. Leave the player body where it is.
+	/// Connects a client to a character or redirects a logged out ConnectedPlayer.
 	/// </summary>
-	/// <param name="conn">connection whose ghost is being spawned</param>
-	/// <param name="playerControllerId">ID of player whose ghost should be spawned</param>
-	/// <returns>the gameobject of the ghost</returns>
-	public static void SpawnPlayerGhost(NetworkConnection conn, short playerControllerId)
+	/// <param name="conn">The client's NetworkConnection. If logged out the playerlist will return an invalid connectedplayer</param>
+	/// <param name="playerControllerId">ID of the client player to be transfered. If logged out it's empty.</param>
+	/// <param name="newBody">The character gameobject to be transfered into.</param>
+	/// <param name="oldBody">The old body of the character.</param>
+	/// <param name="eventType">Event type for the player sync.</param>
+	public static void TransferPlayer(NetworkConnection conn, short playerControllerId, GameObject newBody, GameObject oldBody, EVENT eventType)
 	{
+
 		var connectedPlayer = PlayerList.Instance.Get(conn);
-		GameObject body = connectedPlayer.GameObject;
-		JobType savedJobType = body.GetComponent<PlayerScript>().JobType;
-		GameObject ghost = CreateGhost(connectedPlayer, savedJobType);
-		PlayerList.Instance.UpdatePlayer(conn, ghost);
-		NetworkServer.ReplacePlayerForConnection(conn, ghost, playerControllerId);
-		TriggerEventMessage.Send(ghost, EVENT.GhostSpawned);
-		if (connectedPlayer.Script.PlayerSync != null) {
-			connectedPlayer.Script.PlayerSync.NotifyPlayers(true);
+		if (connectedPlayer == ConnectedPlayer.Invalid)
+		{
+			//conn = new NetworkConnection();
+			ConnectedPlayer loggedOffPlayer = PlayerList.Instance.UpdateLoggedOffPlayer(newBody, oldBody);
+			conn = loggedOffPlayer.Connection;
+			NetworkServer.ReplacePlayerForConnection(conn, newBody, 0); // <- bajillion warnings
```

Comment:
These three lines are the part where I can't wrap my head around, the purpose is to leave a clientless body. Both dummy creation and body transfer while logged out end up here. The new body was already created but if I dont call for `NetworkServer.ReplacePlayerForConnection()`, the new mob just starts runtiming to hell and back. If I DO call for this function, unity gets 1k warnings that even lag the editor for half a second.
Need help!

Reply:
What are the warnings?

Reply:
![image](https://user-images.githubusercontent.com/701959/56870714-fc239e80-69e9-11e9-99b7-3d41bfaca8d3.png)
its a spam of these bad boys, all the same

Reply:
So its all coming from here: https://github.com/unitystation/unitystation/blob/2f53e62956dbfa191cb6b3be42b0bff2c2084962/UnityProject/Assets/Scripts/Messages/Server/ServerMessage.cs#L24

Interesting....

Reply:
See if anything in this thread helps: https://forum.unity.com/threads/reconnect-client-to-server.436033/